### PR TITLE
OCPBUGS-43520: Limit GCP API firewall rule for internal clusters

### PR DESCRIPTION
**Pull request by bfournie**
For internal published clusters, the API firewall rule source should be limited to the machine network CIDR.

### CODE REVIEW

**Review by barbacbd — COMMENTED** [review]
/lgtm

File: pkg/infrastructure/gcp/clusterapi/firewallrules.go [hunk 1]
```diff
@@ -295,11 +295,17 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
...
 	firewallName = fmt.Sprintf("%s-api", in.InfraID)
 	srcTags = []string{}
 	targetTags = []string{masterTag}
-	srcRanges = []string{}
+	if in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy {
+		srcRanges = []string{}
+	} else {
+		// For Internal, limit the source to the machineCIDR
+		srcRanges = []string{machineCIDR}
+	}
```
Comment (patrickdillon) [inline pkg/infrastructure/gcp/clusterapi/firewallrules.go:308]: Need to check mixed publishing as well, and we have a function for that:

https://github.com/openshift/installer/blob/e8f8bf91b0dc2c0488485bd0777ed7f6c658f771/pkg/types/installconfig.go#L581-L593

```suggestion
	srcRanges = []string{}
	if !in.InstallConfig.Config.PublicAPI() {
             srcRanges = append(srcRanges, machineCIDR)
```

Comment (bfournie) [reply pkg/infrastructure/gcp/clusterapi/firewallrules.go]: Thanks. Updated.

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Pull request merged**